pyworklistserver/worklist.py

```diff
@@ -1,108 +1,170 @@
+            return ''.join(random.choices(' ' + string.ascii_uppercase + string.ascii_lowercase + string.digits + __RUSSIAN, k=length))
+        elif (40 <= r < 60) and (user_config.greek_enabled == 1): # GREEK
+            return ''.join(random.choices(' ' + string.ascii_uppercase + string.ascii_lowercase + string.digits + __GREEK, k=length))
+        elif (60 <= r < 80) and (user_config.japanese_enabled == 1): # JAPANESE
+            return ''.join(random.choices(' ' + string.ascii_uppercase + string.ascii_lowercase + string.digits + __JAPANESE, k=length))
+        elif (80 <= r) and (user_config.korean_enabled == 1): # KOREAN
+            return ''.join(random.choices(' ' + string.ascii_uppercase + string.ascii_lowercase + string.digits + __KOREAN, k=length))
+
+    """ Create a random string of specified length containing some non-ascii characters """
+    return ''.join(random.choices(' ' + string.ascii_uppercase + string.ascii_lowercase + string.digits + __NONASCII, k=length))
+
+def _clean_unicode_string(length):
+    """ Create a random string of specified length containing some non-ascii characters """
+    return ''.join(random.choices(' ' + string.ascii_uppercase + string.ascii_lowercase + string.digits + __NONASCII, k=length))
+
+
+def _extend_with_random_to_length(text, length, _None_string_func):
+    """ Extend a string with random characters up to the given length """
+    if _None_string_func() == 1:
+        return None
+    if length == 0:
+        return ''
+    return text + _random_unicode_string(length - len(text))
+
+def _random_person_name(max_len_per_name):
+    """ Create a random person name separated by ^ character """
+    return '^'.join([_random_unicode_string(max_len_per_name) for _ in range(1, random.randrange(2, 4))])
+
+def _random_number_string(max_len):
+    """ Create a random string containing numeric values only """
+    return ''.join(random.choices(string.digits, k=random.randrange(1, max_len)))
+
+def _random_dicom_date_after_1900():
+    """ Return a string containing a random date on DICOM format """
+    start = datetime.date(1900, 1, 1)
+    now = datetime.date.today()
+    maxdays = (now - start)
+    randomdays = random.randrange(0, maxdays.days)
+    date = start + datetime.timedelta(days=randomdays)
+    return date.strftime('%Y%m%d')
+
+def _random_dicom_time():
+    """ Return a string containing a random time point on DICOM format """
+    hour = random.randrange(0, 24)
+    minutes = random.randrange(0, 59)
+    seconds = random.randrange(0, 59)
+    return '{:02}{:02}{:02}'.format(hour, minutes, seconds)
+
+
+
+_VIVID_HACK_MAX_PERSON_NAME = 64
+
+class RandomWorklist:
+    """ Generator for random worklists """
+    def __init__(self, specificCharSet, config_values):
+        self._specific_charset = specificCharSet
+        self._config_values = config_values
+        self._fault_provider = fault_provider.FaultProvider(config_values)
+
+
+    def get_random_worklist(self):
```

Comment:
This function is getting complicated. Please split out a separate class that is responsible for generating random strings. This way you can call

    string_generator._get_random_string(64)

and the string generator takes care of the randomization. The string generator could take a config struct as input to its constructor.

Reply:
fixed in issue #9